lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -3654,7 +3654,13 @@ static bool DominateAllUsers(Instruction *I, Value *V, DominatorTree *DT) {
   if (&F->getEntryBlock() == I->getParent())
     return true;
 
-  return DominateAllUsersDom(I, V, DT);
+  if (!DT) {
+    DominatorTree TempDT;
+    TempDT.recalculate(*F);
```

Comment:
What will happen when V is a global variable which could used in other functions?

Reply:
That is a good point. As stated, I didn't expect global variables to wind up here. ```DominateAllUsersDom``` currently asserts if the functions don't match. It should continue instead to handle globals correctly.